src/langchain_google_spanner/document_loader.py

```diff
@@ -0,0 +1,401 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import json
+from dataclasses import dataclass
+from typing import Any, Dict, Iterator, List, Union
+
+from google.cloud.spanner import Client, KeySet  # type: ignore
+from google.cloud.spanner_admin_database_v1.types import DatabaseDialect  # type: ignore
+from google.cloud.spanner_v1.data_types import JsonObject  # type: ignore
+from langchain_community.document_loaders.base import BaseLoader
+from langchain_core.documents import Document
+
+OPERATION_TIMEOUT_SECONDS = 240
+MUTATION_BATCH_SIZE = 1000
+
+CONTENT_COL_NAME = "page_content"
+METADATA_COL_NAME = "langchain_metadata"
+
+
+@dataclass
+class Column:
+    name: str
+    data_type: str
+    nullable: bool = True
+
+
+def _load_row_to_doc(
+    format: str,
+    content_columns: List[str],
+    metadata_columns: List[str],
+    metadata_json_column: str,
+    row: Dict,
+) -> Document:
+    page_content = ""
+    if format == "text":
+        page_content = " ".join(str(row[c]) for c in content_columns if c in row)
+    elif format == "YAML":
+        page_content = "\n".join(
+            f"{c}: {str(row[c])}" for c in content_columns if c in row
+        )
+    elif format == "JSON":
+        j = {}
+        for c in content_columns:
+            if c in row:
+                j[c] = row[c]
+        page_content = json.dumps(j)
+    elif format == "CSV":
+        page_content = ", ".join(str(row[c]) for c in content_columns if c in row)
+    if not page_content:
```

Comment:
nit: add else with exception ValueError format type doesn't exist. 
I think we can give them empty docs

Reply:
If the format type doesn't exist, the exception would be thrown during init.

Reply:
Removed the `if c in row` check. This way, the loader will throw a keyError automatically.